design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,195 @@
+---
+Service authentication with vault for RGW
+target-version: release-1.14
+---
+
+# Feature Name
+
+Service account authentication with vault for RGW.
+
+## Summary
+
+When `Vault` is set up in Kubernetes environment, it can be authenticated using service account token. The OSD encryption already supports this feature and adding same for RGW. It uses same set of  config option in `ConnectionDetails` for `KeyManagementServiceSpec`.
+
+### Goals
+
+Configure RGW with service account authentication with Vault with help of vault agent.
+
+### Non-Goals
+
+There are different ways RGW can authenticate with help of [vault agent](https://docs.ceph.com/en/latest/radosgw/vault/#vault-agent), but only service account authentication is supported here.
+
+## Proposal details
+
+The [vault agent](https://developer.hashicorp.com/vault/docs/agent-and-proxy/agent#vault-stanza) can be configure with help of following annotation in `CephObjectStore CRD` :
+
+- vault.hashicorp.com/agent-inject --> brings vault agent with rgw pod
+- vault.hashicorp.com/agent-configmap -->  config map containing settings for vault agent pod
+- vault.hashicorp.com/tls-secret --> the name of kubernetes secret containing TLS cert to communicate with vault server.
+
+Other settings are similar to the configuration mentioned in [here](/Documentation/Storage-Configuration/Advanced/key-management-system/#kubernetes-based-authentication)
+
+The `Ceph Object Store` CRD will look as :
+
+```yaml
+spec:
+  gateway:
+  annotations:
+    vault.hashicorp.com/agent-inject: 'true'
+    vault.hashicorp.com/agent-configmap: <ceph-object-store>-vault-agent-cm
+  security:
+    kms:
+      connectionDetails:
+          KMS_PROVIDER: vault
+          VAULT_ADDR: http://127.0.0.1:8100
+          VAULT_BACKEND_PATH: rook
+          VAULT_SECRET_ENGINE: kv
+          VAULT_AUTH_METHOD: kubernetes
+          VAULT_AUTH_KUBERNETES_ROLE: rook-ceph
+
+---
+kind: ConfigMap
```

Comment:
Does the user create this configmap, or does Rook generate it? We need to be clear about what the admin needs to create, and what Rook will generate.

Reply:
IMO since there are a lot of options I prefer the user to configure cm than by Rook Operator. We can provide the sample configuration(minimal configuration) as above